### PR TITLE
general fixes and checkpoint state accessors

### PR DESCRIPTION
**Pull request by adlrocha**
This PR: 
- Fixes subnet serialization.
- Includes a explicit check to HAMT and AMT functions for when there are no errors but the key doesn't exist.
- Implements state accessors for checkpoints. 
- Implements CID computation for checkpoints.

### CODE REVIEW

File: subnetactor/types.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ type Votes struct {
 
 // ConsensusType defines the types of consensus supported
 // by subnets.
-type ConsensusType int64
+type ConsensusType uint64
```
Comment (dnkolegov) [inline subnetactor/types.go:40]: We use this type and the protocols below at least in two repos/packages: Eudico/Lotus and here. Should we store it only in one package and import it?

Comment (dnkolegov) [reply subnetactor/types.go:40]: `IpcGatewayAddr` must be `IPCGatewayAddr`

File: gateway/types.go [hunk 2]
```diff
@@ -92,6 +111,25 @@ type Checkpoint struct {
...
+	h, err := mh.Sum(buf.Bytes(), abi.HashFunction, -1)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	return cid.NewCidV1(abi.CidBuilder.GetCodec(), h), nil
+}
+
 func (c *Checkpoint) CrossMsgMeta(from, to *sdk.SubnetID) (*CrossMsgMeta, bool) {
 	for i, m := range c.Data.CrossMsgs {
```
Comment (dnkolegov) [inline gateway/types.go:134]: Why are we sure that CrossMsgs field is not nil?

File: gateway/types.go [hunk 3]
```diff
@@ -23,10 +26,26 @@ type Subnet struct {
...
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return NewCheckpoint(st.NetworkName, epoch), nil
+	}
+	return ch, nil
+}
+
 type StorableMsg struct {
```
Comment (dnkolegov) [inline gateway/types.go:49]: Just a note: Storable sounds like an interface name

**Review by dnkolegov — APPROVED** [review]
LGTM!
I added minor comments related to naming and formatting. 
It would be good to add a golangci rule to check that imported packages are properly sorted.

File: subnetactor/types.go (hunk 1 above)
Comment (adlrocha) [reply subnetactor/types.go:41]: Fixed

File: gateway/types.go (hunk 2 above)
Comment (adlrocha) [reply gateway/types.go:119]: We are not, Alex worked on that and I haven't reviewed those methods yet. Will fix this one now.

**Pull request merged**